### PR TITLE
Properly activate timestamp fork

### PR DESCRIPTION
**Pull request by flcl42**
Fixes hive test "Pre-Merge Fork Number > 0 (Shanghai)"

## Changes

- Prioritize timestamp forks if block timestamp is bigger than the first timestamp-based fork value
- So it fixes corner case when timestamp matches some timestamp fork and block number is exact number of block number only fork

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

#### Requires documentation update

- [ ] Yes
- [x] No

#### Requires explanation in Release Notes

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecBasedSpecProvider.cs [hunk 1]
```diff
@@ -276,6 +278,15 @@ public IReleaseSpec GetSpec(ForkActivation activation)
                 {
                     if (_logger.IsWarn) _logger.Warn($"Chainspec file is misconfigured! Timestamp transition is configured to happen before the last block transition.");
                 }
+
+                if (_firstTimestampActivation.Value.Timestamp <= activation.Timestamp)
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecBasedSpecProvider.cs:282]: would be a bit simpler if in the `if` you only choose which array to look:
something like:
```suggestion
                (ForkActivation Activation, ReleaseSpec Spec)[] consideredTimestamps =   _firstTimestampActivation.Value.Timestamp <= activation.Timestamp ?_timestampOnlyTransitions : _transitions;
```

**Review by smartprogrammer93 — APPROVED** [review]
This works. keep in mind in normal circumstances this should never happen. We cant have a fork using blocknumber after or at the samr time as a timestamp activation. But given this is a hive test (can also apply to testnets where they enable EIPs at the same block nimnber and timestamp of the first block or something.)  Its all good

**Review by LukaszRozmej — APPROVED** [review]

**Comment by flcl42** [comment]
> This works. keep in mind in normal circumstances this should never happen. We cant have a fork using blocknumber after or at the samr time as a timestamp activation. But given this is a hive test (can also apply to testnets where they enable EIPs at the same block nimnber and timestamp of the first block or something.) Its all good

Yeah, you got it correctly. I almost asked the hive to change their tests(which is may be not very convenient to always keep such case in mind for nethermind), but luckily found that solution can be also a small optimization: we choose from timestamp forks only now in most cases.
